registration-processor/pre-processor/registration-processor-cmd-validator-stage/src/main/java/io/mosip/registration/processor/stages/cmdvalidator/DeviceValidator.java

```diff
@@ -62,81 +87,148 @@
-				deviceValidateHistoryRequest.setDigitalId(registeredDevice.getDigitalId());
-				deviceValidateHistoryRequest.setTimeStamp(regOsi.getPacketCreationDate());
-				RequestWrapper<DeviceValidateHistoryRequest> request = new RequestWrapper<>();
-
-				request.setRequest(deviceValidateHistoryRequest);
-				request.setId(deviceValidateHistoryId);
-				request.setMetadata(null);
-				request.setVersion("1.0");
-				DateTimeFormatter format = DateTimeFormatter.ofPattern(env.getProperty(DATETIME_PATTERN));
-				LocalDateTime localdatetime = LocalDateTime
-						.parse(DateUtils.getUTCCurrentDateTimeString(env.getProperty(DATETIME_PATTERN)), format);
-				request.setRequesttime(localdatetime);
-
-				DeviceValidateHistoryResponse deviceValidateResponse;
-
-				ResponseWrapper<?> responseWrapper = (ResponseWrapper<?>) registrationProcessorRestService
-						.postApi(ApiName.DEVICEVALIDATEHISTORY, "", "", request, ResponseWrapper.class);
-				deviceValidateResponse = mapper.readValue(mapper.writeValueAsString(responseWrapper.getResponse()),
-						DeviceValidateHistoryResponse.class);
-				if (responseWrapper.getErrors() == null || responseWrapper.getErrors().isEmpty()) {
-					if (!deviceValidateResponse.getStatus().equalsIgnoreCase(VALID)) {
-						throw new ValidationFailedException(StatusUtil.DEVICE_VALIDATION_FAILED.getMessage(),
-								StatusUtil.DEVICE_VALIDATION_FAILED.getCode());
-					}
-				} else {
-					List<ErrorDTO> error = responseWrapper.getErrors();
-					regProcLogger.error("validate call ended for registrationId {} with error data : {}",
-							registrationId, error.get(0).getMessage());
-					throw new BaseCheckedException(
-							error.get(0).getMessage() + " for " + deviceDetails.getDigitalId().getType(),
-							StatusUtil.DEVICE_VALIDATION_FAILED.getCode());
-
+	public void validate(RegOsiDto regOsi,String process, String registrationId)
+			throws JsonProcessingException, IOException, BaseCheckedException, ApisResourceAccessException, JSONException {
+
+		BiometricRecord biometricRecord = packetManagerService.getBiometricsByMappingJsonKey(registrationId,
+				MappingJsonConstants.INDIVIDUAL_BIOMETRICS, process,
+				ProviderStageName.CMD_VALIDATOR);
+		List<BIR> birs=biometricRecord.getSegments();
+		List<JSONObject> payloads=new ArrayList<>();
+		for(BIR bir: birs) {
+			if(bir.getOthers()!=null) {
+			for(Entry entry: bir.getOthers()) {
+				if(entry.getKey().equals("PAYLOAD")) {
+					payloads.add(new JSONObject(entry.getValue()));				
 				}
-				regProcLogger.debug("validate call ended for registrationId {} with response data : {}", registrationId,
-						JsonUtil.objectMapperObjectToJson(deviceValidateResponse));
 			}
+			}
+		}
+		if(!payloads.isEmpty()) {
+		for(JSONObject payload :payloads) {
+			if(!validateSignature(payload) ||
+			!validateTimeStamp(payload,regOsi.getPacketCreationDate()) ||
+			!isDeviceHotlisted(payload.getString("deviceCode"),payload.getString("timestamp"))
+			) {
```

Comment:
For all the above 3 validation method, since you are directly throwing the exception on validation failure, remove this if condition and make the return to void for 3.